zomes/countersigning/coordinator/src/lib.rs

```diff
@@ -0,0 +1,186 @@
+use countersigning_integrity::*;
+use hdk::prelude::*;
+
+#[repr(u8)]
+pub enum Roles {
+    Vendor = 1,
+    Taster = 2,
+}
+
+impl From<Roles> for Role {
+    fn from(value: Roles) -> Self {
+        Role(value as u8)
+    }
+}
+
+#[hdk_extern]
+fn init() -> ExternResult<InitCallbackResult> {
+    let mut fns = BTreeSet::new();
+    fns.insert((zome_info()?.name, "recv_remote_signal".into()));
+    let functions = GrantedFunctions::Listed(fns);
+    create_cap_grant(CapGrantEntry {
+        tag: "".into(),
+        // empty access converts to unrestricted
+        access: ().into(),
+        functions,
+    })?;
+
+    Ok(InitCallbackResult::Pass)
+}
+
+#[hdk_extern]
+fn recv_remote_signal(signal: Signals) -> ExternResult<()> {
+    // TODO we don't know who the remote signal came from?!
+    emit_signal(signal)
+}
+
+#[hdk_extern]
+fn initiator_hello() -> ExternResult<String> {
+    Ok("Hello!".to_string())
+}
+
+#[hdk_extern]
+fn participant_hello() -> ExternResult<()> {
+    create_link(
+        hash_entry(EntryTypes::ParticipantBase(ParticipantBase))?,
+        agent_info()?.agent_initial_pubkey,
+        LinkTypes::Participant,
+        (),
+    )?;
+
+    Ok(())
+}
+
+#[hdk_extern]
+fn list_participants() -> ExternResult<Vec<AgentPubKey>> {
+    let links = get_links(
+        GetLinksInputBuilder::try_new(
+            hash_entry(EntryTypes::ParticipantBase(ParticipantBase))?,
+            LinkTypes::Participant,
+        )?
+        .build(),
+    )?;
+    Ok(links
+        .into_iter()
+        .map(|link| link.target.clone().try_into().unwrap())
+        .collect())
+}
+
+#[hdk_extern]
+fn start_two_party(with_other: AgentPubKey) -> ExternResult<PreflightResponse> {
+    let my_agent_info = agent_info()?;
+
+    let entry = ImportantAgreement {
+        best_ice_cream_flavour: "strawberry".to_string(),
```

Comment:
Clearly it's chocolate

Reply:
I went down this route of creating some fun test data, then really struggled to pick. I wanted chocolate, banana, mint choc-chip and several more.... I really like ice cream :)